pkg/manifests/manifest_integration_test.go

```diff
@@ -0,0 +1,216 @@
+	require.ErrorContains(t, err, "downloading manifest 'k8s.io/examples/application/nginx-app.yaml': executing request: Get \"k8s.io/examples/application/nginx-app.yaml\": unsupported protocol scheme \"")
+	assert.Equal(t, 0, len(manifestPaths))
+}
+
+func TestGetAllImages(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	expectedContainerImages := []image.ContainerImage{
+		{
+			Name: "nginx:latest",
+		},
+		{
+			Name: "node:14",
+		},
+		{
+			Name: "mysql:5.7",
+		},
+		{
+			Name: "redis:6.0",
+		},
+		{
+			Name: "custom-api:1.2.3",
+		},
+		{
+			Name:           "quay.io/podman/hello",
+			SupplyChainKey: "sample-key",
+		},
+	}
+	sort.Slice(expectedContainerImages, func(i, j int) bool {
+		return expectedContainerImages[i].Name < expectedContainerImages[j].Name
+	})
+
+	localManifestSourceDir := filepath.Join(ctx.ImageConfigDir, "kubernetes", "manifests")
+	err := os.MkdirAll(localManifestSourceDir, os.ModePerm)
+	require.NoError(t, err)
+
+	localSampleManifestPath := filepath.Join("testdata", "sample-crd.yaml")
+	err = fileio.CopyFile(localSampleManifestPath, filepath.Join(localManifestSourceDir, "sample-crd.yaml"), fileio.NonExecutablePerms)
+	require.NoError(t, err)
+
+	ctx.ImageDefinition = &image.Definition{
+		EmbeddedArtifactRegistry: image.EmbeddedArtifactRegistry{
+			ContainerImages: []image.ContainerImage{
+				{
+					Name:           "quay.io/podman/hello",
+					SupplyChainKey: "sample-key",
+				},
+			},
+		},
+		Kubernetes: image.Kubernetes{
+			Manifests: image.Manifests{
+				URLs: []string{
+					"https://k8s.io/examples/application/nginx-app.yaml",
+				},
+			},
+		},
+	}
+
+	// Test
+	containerImages, err := GetAllImages(ctx)
+	sort.Slice(containerImages, func(i, j int) bool {
+		return containerImages[i].Name < containerImages[j].Name
+	})
+
+	// Verify
+	require.NoError(t, err)
+	assert.Equal(t, expectedContainerImages, containerImages)
+}
+
+func TestGetAllImagesInvalidLocalManifest(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	localManifestSourceDir := filepath.Join(ctx.ImageConfigDir, "kubernetes", "manifests")
+	err := os.MkdirAll(localManifestSourceDir, os.ModePerm)
+	require.NoError(t, err)
+
+	localSampleManifestPath := filepath.Join("testdata", "invalid-crd.yaml")
+	err = fileio.CopyFile(localSampleManifestPath, filepath.Join(localManifestSourceDir, "sample-crd.yaml"), fileio.NonExecutablePerms)
+	require.NoError(t, err)
+
+	// Test
+	_, err = GetAllImages(ctx)
+
+	// Verify
+	require.ErrorContains(t, err, "error reading manifest error unmarshalling manifest yaml")
+}
+
+func TestGetAllImagesInvalidURL(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition.Kubernetes.Manifests.URLs = []string{
+		"k8s.io/examples/application/nginx-app.yaml",
+	}
+
+	// Test
+	_, err := GetAllImages(ctx)
+
+	// Verify
+	require.ErrorContains(t, err, "error downloading manifests: downloading manifest")
```

Comment:
Can we include more of the error? It's not quite clear why it fails aside from the name of the test itself.

Reply:
Just tested, unfortunately the order comes back different every time (I assume cause it's a map at first)